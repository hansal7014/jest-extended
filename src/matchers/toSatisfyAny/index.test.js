import matcher from './';

expect.extend(matcher);

let isEven = el => el % 2 === 0;
let isOdd = el => el % 2 === 1;

describe('.toSatisfyAny', () => {
  test('passes when any value satisfies predicate', () => {
    expect([1, 2, 4, 6]).toSatisfyAny(isOdd);
    expect([2, 4, 6, 8]).toSatisfyAny(isEven);
    expect([11]).toSatisfyAny(isOdd);
    expect([10]).toSatisfyAny(isEven);
  });

  test('fails when all values do not satisfy the predicate', () => {
    expect(() => expect([1, 3, 7, 5]).toSatisfyAny(isEven)).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toSatisfyAny', () => {
  test('passes when all values do not satisfy the predicate', () => {
    expect([2, 4, 6, 8]).not.toSatisfyAny(isOdd);
    expect([1, 5, 3, 7, 9]).not.toSatisfyAny(isEven);
  });

  test('fails when any value satisfies predicate', () => {
    expect(() => expect([1, 2, 3, 4]).not.toSatisfyAny(isOdd)).toThrowErrorMatchingSnapshot();
  });
});
