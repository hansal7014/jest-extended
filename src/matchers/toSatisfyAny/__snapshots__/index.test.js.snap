// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toSatisfyAny fails when any value satisfies predicate 1`] = `
"<dim>expect(</><red>received</><dim>).not.toSatisfyAny(</><green>expected</><dim>)</>

Expected array to not satisfy predicate for any value:
  <green>[Function isOdd]</>
Received:
  <red>[1, 2, 3, 4]</>"
`;

exports[`.toSatisfyAny fails when all values do not satisfy the predicate 1`] = `
"<dim>expect(</><red>received</><dim>).toSatisfyAny(</><green>expected</><dim>)</>

Expected array to satisfy predicate for any value:
  <green>[Function isEven]</>
Received:
  <red>[1, 3, 7, 5]</>"
`;
